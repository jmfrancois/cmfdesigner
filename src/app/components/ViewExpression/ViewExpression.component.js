import React from 'react';
import PropTypes from 'prop-types';
import { cmfConnect, Inject } from '@talend/react-cmf';
import ViewFunctionParam from '../ViewFunctionParam';


function ViewExpression(props) {
	const hasKey = (props.item.key && props.item.name !== props.item.key);
	const hasParams = props.item.params && props.item.params.length > 0;
	return (
		<div>
			<h1>Expression: {hasKey ? props.item.key : props.item.name}</h1>
			<div>
				<ul>
					{hasKey && <li>function name: {props.item.name}</li>}
					{(hasParams) && (
						<li>Params: <ViewFunctionParam params={props.item.params} /></li>
					)}
				</ul>
			</div>
			<Inject component="FileAnalytics" path={props.item.path} />
		</div>
	);
}

ViewExpression.propTypes = {
	item: PropTypes.shape({
		key: PropTypes.string,
		name: PropTypes.string,
		params: PropTypes.array,
		path: PropTypes.string,
	}),
};
ViewExpression.defaultProps = {
	item: { params: [] },
};
ViewExpression.displayName = 'ViewExpression';

export default cmfConnect({})(ViewExpression);
