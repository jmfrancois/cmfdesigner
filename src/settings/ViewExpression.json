{
  "props": {
    "ViewExpression#default": {
      "itemExpression": "getExpression",
      "renderIfExpression": {
        "id": "cmf.router.matchPath",
        "args": [
            {
                "path": "/expressions/view",
                "isExact": true
            }
        ]
      }
    }
  }
}
