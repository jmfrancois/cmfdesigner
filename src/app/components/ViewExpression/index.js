import ViewExpression from './ViewExpression.component';

export default ViewExpression;
