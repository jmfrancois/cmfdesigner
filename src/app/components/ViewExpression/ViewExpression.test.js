import React from 'react';
import { shallow } from 'enzyme';

import ViewExpression from './ViewExpression.component';

describe('ViewExpression', () => {
	it('should render', () => {
		const wrapper = shallow(
			<ViewExpression />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
