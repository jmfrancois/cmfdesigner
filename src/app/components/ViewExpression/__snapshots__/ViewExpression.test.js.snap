// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewExpression should render 1`] = `
<div>
  <h1>
    Expression: 
  </h1>
  <div>
    <ul>
      
    </ul>
  </div>
  <Inject
    component="FileAnalytics"
    path={undefined}
  />
</div>
`;
